<?xml version="1.0" ?>
<launch>

    <arg name="markerSize"      default="0.1"/>  <!-- in m. con valori >= 0.5 la terna si ritrova al centro del marker (OK), mentre con valori <=0.5, incluso 0.08 (vera dimensione del marker), c'è un offset -->
    <arg name="eye"             default="left"/>-->
    <arg name="camera_frame"    default="camera_link_turtlebot3_frame"/> <!-- forse è più corretto turtlebot/camera_link, ma credo sia ininfluente visto che funziona anche così-->
    <arg name="marker_frame"    default="aruco_marker_frame_turtlebot"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->

    <node pkg="tf" type="static_transform_publisher" name="turtlebot3_camera_tf" args="0 0 0 -1.5707 0 -1.5707 /camera_link /camera_link_turtlebot3_frame 100"/>

    <!--<node pkg="aruco_ros" type="single" name="aruco_single"> -->
    <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher_turtlebot">
        <!-- <remap from="/camera_info" to="/stereo/$(arg eye)/camera_info" />
             <remap from="/image" to="/stereo/$(arg eye)/image_rect_color" /> -->

        <remap from="/camera_info" to="/camera_turtlebot/camera_info" />     <!-- "/turtlebot/camera/camera_info"-->
        <remap from="/image" to="/camera_turtlebot/image_raw" />              <!-- "/turtlebot/camera/image_raw" -->
        
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <!--<param name="camera_frame"       value="stereo_gazebo_$(arg eye)_camera_optical_frame"/> -->
        <param name="camera_frame"       value="$(arg camera_frame)"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

</launch>
